[Bug]: 0.14.0 Build Error

Following the README Dependencies and Installing instructions results in the below error after running `make`:

```
192.0 Boss/Mod/Initiator.cpp:1:9: fatal error: commit_hash.h: No such file or directory
192.0     1 | #include"commit_hash.h"
192.0       |         ^~~~~~~~~~~~~~~
192.0 compilation terminated.
```

Interestingly, it seems this file should exist when I look at the top of the stack trace: `0.110 Using existing ./commit_hash.h`.

Build Environment:
- debian:bullseye-slim
- clboss 0.14.0 with a clean working tree

I have included the Dockerfile steps below for reference:
```
FROM debian:bullseye-slim AS clboss

RUN apt-get update -qq && \
    apt-get install -qq -y --no-install-recommends \
        autoconf-archive \
        automake \
        build-essential \
        git \
        libcurl4-gnutls-dev \
        libev-dev \
        libsqlite3-dev \
        libtool \
        pkg-config

COPY clboss/. /tmp/clboss
WORKDIR /tmp/clboss
RUN autoreconf -i
RUN ./configure
RUN make
RUN make install
RUN strip /usr/local/bin/clboss
```

These same build steps above have worked for at least the last several releases of clboss. Looking at the README, it doesn't seem anything should have changed breaking these build steps. Is there perhaps something that did change that wasn't updated in the README to get this build to work for 0.14.0?

--- a/Makefile.am
+++ b/Makefile.am
@@ -679,8 +679,10 @@ create-tarball: create-tarball.in Makefile
 BUILT_SOURCES = commit_hash.h
 
 commit_hash.h: FORCE
-	@if test -e $(srcdir)/.git/logs/HEAD; then \
-		if [ ! -f $(srcdir)/commit_hash.h ] || [ $(srcdir)/.git/logs/HEAD -nt $(srcdir)/commit_hash.h ]; then \
+	@if git -C $(srcdir) rev-parse --is-inside-work-tree > /dev/null 2>&1; then \
+		CURRENT_HASH="$$(git -C $(srcdir) rev-parse HEAD)"; \
+		if [ ! -f $(srcdir)/commit_hash.h ] || \
+		   [ "$$(grep -oE '[0-9a-f]{40}' $(srcdir)/commit_hash.h)" != "$$CURRENT_HASH" ]; then \
 			echo "Regenerating $(srcdir)/commit_hash.h..."; \
 			$(SHELL) $(srcdir)/generate_commit_hash.sh; \
 		else \

